src/components/project_view/submission_detail/mutation_suite_result.vue

```diff
@@ -0,0 +1,797 @@
+  submission!: Submission;
+
+  @Prop({required: true, type: String})
+  fdbk_category!: FeedbackCategory;
+
+  @Watch('submission')
+  on_submission_change(new_value: Submission, old_value: Submission) {
+    this.d_submission = new_value;
+  }
+
+  @Watch('fdbk_category')
+  on_fdbk_category_change(new_value: FeedbackCategory, old_value: FeedbackCategory) {
+    this.d_fdbk_category = new_value;
+  }
+
+  @Watch('mutation_test_suite_result')
+  async on_mutation_test_suite_results_change(new_value: MutationTestSuiteResultFeedback,
+                                              old_value: MutationTestSuiteResultFeedback) {
+    this.d_mutation_test_suite_result = JSON.parse(JSON.stringify(new_value));
+    await this.get_results();
+  }
+
+  readonly CorrectnessLevel = CorrectnessLevel;
+  readonly BugsExposedFeedbackLevel = BugsExposedFeedbackLevel;
+  readonly ReturnCodeCorrectness = ReturnCodeCorrectness;
+
+  d_setup_stdout_content: string | null = null;
+  d_setup_stderr_content: string | null = null;
+  d_student_test_names_stdout_content: string | null = null;
+  d_student_test_names_stderr_content: string | null = null;
+  d_validity_check_stdout_content: string | null = null;
+  d_validity_check_stderr_content: string | null = null;
+  d_grade_buggy_stdout_content: string | null = null;
+  d_grade_buggy_stderr_content: string | null = null;
+
+  d_submission: Submission | null = null;
+  d_fdbk_category: FeedbackCategory = FeedbackCategory.past_limit_submission;
+  d_mutation_test_suite_result: MutationTestSuiteResultFeedback | null = null;
+
+  d_show_buggy_implementations_output = false;
+  d_show_student_test_names_output = false;
+  d_show_validity_check_output = false;
+
+  d_load_grade_buggy_output = false;
+  d_load_student_test_names_output = false;
+  d_load_validity_check_output = false;
+
+  d_setup_stdout_loaded = false;
+  d_setup_stderr_loaded = false;
+  d_student_test_names_stdout_loaded = false;
+  d_student_test_names_stderr_loaded = false;
+  d_validity_check_stdout_loaded = false;
+  d_validity_check_stderr_loaded = false;
+  d_grade_buggy_stdout_loaded = false;
+  d_grade_buggy_stderr_loaded = false;
+  d_mutation_test_suite_result_output_size: ResultOutput.MutationTestSuiteResultOutputSize
+                                            | null = null;
+
+  async created() {
+    this.d_fdbk_category = this.fdbk_category;
+    this.d_submission = this.submission;
+    this.d_mutation_test_suite_result = this.mutation_test_suite_result;
+    await this.get_results();
+  }
+
+  get show_setup_fieldset(): boolean {
+    return this.d_mutation_test_suite_result!.fdbk_settings.show_setup_stdout
+           || this.d_mutation_test_suite_result!.fdbk_settings.show_setup_stderr
+           || this.d_mutation_test_suite_result!.setup_return_code !== null
+           || (this.d_mutation_test_suite_result!.setup_timed_out !== null
+               && this.d_mutation_test_suite_result!.setup_timed_out!);
```

Comment:
When I try this approach I get the warning: "This type is not allowed in the operand for the '!' operator because it could be null. Only booleans are allowed."

Reply:
Huh, It was fine when I tried it. I might have done something slightly differently. Go ahead and leave it as is then